graphql_compiler/compiler/subclass.py

```diff
@@ -54,6 +45,8 @@ def compute_subclass_sets(schema, type_equivalence_hints=None):
         else:
             raise AssertionError(u'Unexpected type {}'.format(type(equivalent_type)))
 
-    # If B subclasses A, and C subclasses B, then C subclasses A
-    _add_transitive_closure(subclass_set)
+    # The inheritance structure in the GraphQL schema is flat. Union type encompass all of the
+    # object type subclasses of their equivalent object type and cannot encompass other
+    # union types. Interface types are implemented by all their object type subclasses and cannot
```

Comment:
Consider: A is equivalent to union_A, that contains B, which is equivalent to union_B, which contains C.

Reply:
Then in the GraphQL schema union_A should already encompass C. Suppose that we had an edge E from B to union_A. If union_A did not encompass C already, then I there would be no way to use transverse E from B to C, but there should be because C is a subclass of A.

Reply:
ok

Reply:
I'll add this in the comment discussion in the comment for future reference.